assets/product-form.js

```diff
@@ -10,9 +10,11 @@ if (!customElements.get('product-form')) {
     }
 
     onSubmitHandler(evt) {
+      if (this.dataset.addToCartBehavior == 'go_to_cart') return;
```

Comment:
For the first issue with the cart error, I'm thinking it might be a local environment issue (or a bug that came from elsewhere) since I didn't change any of the actual cart behaviours.

For the second issue, I'm wondering if we should do anything to change that... my thought is, the merchant is looking to effectively take the user right to the cart page and if we were to show an error on the product page, then that would mean we're still making an AJAX call and then sending them there only if there's a success.  This will slow things down a lot compared to what I'm doing right now which is effectively just submitting the form (basically the `no-js` flow).

I do find it interesting that the cart can't auto-adjust though, but I wonder if that's a limitation of the Liquid API.  Like when you go to the cart page, it would make sense for the Liquid to have an awareness that the total quantity is higher than the available quantity.  Because again, if the Liquid doesn't have that available, it's another situation where we'd need to make the check on the cart page when the page load via AJAX and then make adjustments.  This also goes against one of our theme development principles of "don't do DOM adjustments prior to user input".

Reply:
Yeah it's kind of an odd situation. I think that if I can't add the product to the cart because there is none left/I've added them all already, then don't take me to the cart. Let me know on the spot rather than make me wait and load a new page to tell me that 🤔  

It's kind of like a frustrating flow there was with recaptcha where sometimes you'd need to go through captcha verification to just be told the email was already registered or something. Ideally if something is wrong or can't be done, don't make me wait and tell me right away. 

My 🪙 🪙  🙂

Reply:
Yeah, the problem is just that it would cause a delay in all of the cases where the action **is** valid since we'd need to do a round-trip AJAX call and then redirect to the cart anyway.  Since the checkout itself can correct this, I think it makes sense for us to leave this behaviour in because the alternative is too clunky.

Reply:
I think we need to add more logic in here then.
So that if `this.dataset.addToCartBehavior == 'go_to_cart'` is true we run a check on the product quantity available before going to the cart page with a fetch 🤔